code/iaas/auth-logic/src/main/java/io/cattle/platform/iaas/api/auth/integration/ldap/LdapServiceContextPoolFactory.java

```diff
@@ -0,0 +1,64 @@
+package io.cattle.platform.iaas.api.auth.integration.ldap;
+
+import java.util.Hashtable;
+import javax.naming.Context;
+import javax.naming.NamingException;
+import javax.naming.ldap.InitialLdapContext;
+import javax.naming.ldap.LdapContext;
+
+import org.apache.commons.lang3.StringUtils;
+import org.apache.commons.logging.Log;
+import org.apache.commons.logging.LogFactory;
+import org.apache.commons.pool2.PooledObject;
+import org.apache.commons.pool2.PooledObjectFactory;
+import org.apache.commons.pool2.impl.DefaultPooledObject;
+
+public class LdapServiceContextPoolFactory implements PooledObjectFactory<LdapContext> {
+
+    private static final Log logger = LogFactory.getLog(LdapServiceContextPoolFactory.class);
+
+    @Override
+    public PooledObject<LdapContext> makeObject() throws Exception {
+        String username = LdapConstants.SERVICE_ACCOUNT_USER.get();
+        if (StringUtils.isNotBlank(LdapConstants.LDAP_LOGIN_DOMAIN.get())) {
+            username = LdapConstants.LDAP_LOGIN_DOMAIN.get() + '\\' +username;
+        }
+        Hashtable<String, String> props = new Hashtable<>();
+        props.put(Context.SECURITY_AUTHENTICATION, "simple");
+        props.put(Context.SECURITY_PRINCIPAL, username);
+        props.put(Context.SECURITY_CREDENTIALS, LdapConstants.SERVICE_ACCOUNT_PASSWORD.get());
+        props.put(Context.INITIAL_CONTEXT_FACTORY, "com.sun.jndi.ldap.LdapCtxFactory");
+        LdapContext userContext;
+
+        try {
+            String url = "ldap://" + LdapConstants.LDAP_SERVER.get() + ':' + LdapConstants.LDAP_PORT.get() + '/';
+            props.put(Context.PROVIDER_URL, url);
+            if (LdapConstants.TLS_ENABLED.get()) {
+                props.put(Context.SECURITY_PROTOCOL, "ssl");
+            }
+            userContext = new InitialLdapContext(props, null);
+        } catch (NamingException e) {
+            logger.info("Failed to login to ldap user:" + username, e);
+            throw new RuntimeException(e);
+        }
+        return new DefaultPooledObject<>(userContext);
+    }
+
+    @Override
+    public void destroyObject(PooledObject<LdapContext> p) throws Exception {
+        p.getObject().close();
+    }
+
+    @Override
+    public boolean validateObject(PooledObject<LdapContext> p) {
+        return true;
```

Comment:
Validate to handle a situation in which the pooled object connetion times out or becomes bad for some reason.